sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -517,13 +525,56 @@ class GpuDynamicPartitionDataSingleWriter(
       // NOTE: the `zip` here has the effect that will remove an extra `ContiguousTable`
       // added at the end of `splits` because we use `upperBound` to find the split points,
       // and the last split point is the number of rows.
+      val outDataTypes = description.dataColumns.map(_.dataType).toArray
       splits.zip(paths).zipWithIndex.map { case ((split, path), ix) =>
         splits(ix) = null
-        SplitAndPath(split, path, ix)
+        withResource(split) { _ =>
```

Comment:
But SpillableColumnarBatch takes ownership of split. Why are we closing it here too?

Reply:
No unfortunately. `SpillableColumnarBatch` doesn't handle `ContiguousTable` the same as `ColumnarBatch`... I will make it consistent, let me file something official and assign it to me before I forget.

Reply:
I forgot about that.

Reply:
https://github.com/NVIDIA/spark-rapids/issues/8669